OSGi metadata is incorrect

The manifest is incorrectly exporting the org.objenesis package. This package is not included in the mockito jar, so this package needs to be imported rather than exported.

It looks like the metadata is correct, however when running tests in an osgi container, I am unable to mock classes (see error below). I built a custom version of the mockito jar and embedded the byte-buddy and objenesis classes inside the mockito jar (instead of loading them as separate bundles) and the error went away. I'm happy to send a copy of the jar file that I was able to get to work.

This is the error I get if I use the mockito jar file in maven (org.mockito:mockito-core:2.0.70-beta):

`com.proofpoint.governance.agent.tagging.file.sync.test.TaggedFileSyncServiceIntegrationTest STANDARD_ERROR
    org.mockito.exceptions.base.MockitoException:
    Mockito cannot mock this class: interface com.proofpoint.governance.agent.tagging.file.sync.api.ServerTaggedFileSyncService

    Mockito can only mock visible & non-final classes.
    If you're not sure why you're getting this error, please report to the mailing list.

    Underlying exception : java.lang.IllegalStateException: Error invoking java.lang.ClassLoader#findClass

        at com.proofpoint.governance.agent.tagging.file.sync.test.osgi.FileTaggingSyncTestActivator.init(FileTaggingSyncTestActivator.java:14)
        at com.proofpoint.governance.common.osgi.api.BaseActivator.init(BaseActivator.java:36)
        at org.apache.felix.dm.DependencyActivatorBase.start(DependencyActivatorBase.java:75)
        at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:697)
        at org.apache.felix.framework.Felix.activateBundle(Felix.java:2226)
        at org.apache.felix.framework.Felix.startBundle(Felix.java:2144)
        at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1371)
        at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
        at java.lang.Thread.run(Thread.java:745)
    Caused by: java.lang.IllegalStateException: Error invoking java.lang.ClassLoader#findClass
        at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$Resolved.loadClass(ClassInjector.java:401)
        at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection.inject(ClassInjector.java:210)
        at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default$InjectionDispatcher.load(ClassLoadingStrategy.java:204)
        at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default.load(ClassLoadingStrategy.java:119)
        at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:4350)
        at org.mockito.internal.creation.bytebuddy.MockBytecodeGenerator.generateMockClass(MockBytecodeGenerator.java:60)
        at org.mockito.internal.creation.bytebuddy.CachingMockBytecodeGenerator$CachedBytecodeGenerator.generate(CachingMockBytecodeGenerator.java:73)
        at org.mockito.internal.creation.bytebuddy.CachingMockBytecodeGenerator$CachedBytecodeGenerator.getOrGenerateMockClass(CachingMockBytecodeGenerator.java:65)
        at org.mockito.internal.creation.bytebuddy.CachingMockBytecodeGenerator.get(CachingMockBytecodeGenerator.java:28)
        at org.mockito.internal.creation.bytebuddy.ByteBuddyMockMaker.createProxyClass(ByteBuddyMockMaker.java:54)
        at org.mockito.internal.creation.bytebuddy.ByteBuddyMockMaker.createMock(ByteBuddyMockMaker.java:27)
        at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:33)
        at org.mockito.internal.MockitoCore.mock(MockitoCore.java:64)
        at org.mockito.Mockito.mock(Mockito.java:1559)
        at org.mockito.Mockito.mock(Mockito.java:1472)
        ... 9 more
    Caused by: java.lang.NoClassDefFoundError: org/mockito/internal/creation/bytebuddy/MockMethodInterceptor$MockAccess
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$Resolved.loadClass(ClassInjector.java:397)
        ... 23 more
    Caused by: java.lang.ClassNotFoundException: org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$MockAccess
        at java.lang.ClassLoader.findClass(ClassLoader.java:530)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at net.bytebuddy.dynamic.loading.MultipleParentClassLoader.loadClass(MultipleParentClassLoader.java:67)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 30 more`

--- a/src/main/java/org/mockito/internal/creation/bytebuddy/MockBytecodeGenerator.java
+++ b/src/main/java/org/mockito/internal/creation/bytebuddy/MockBytecodeGenerator.java
@@ -61,9 +61,11 @@ class MockBytecodeGenerator {
                       .load(new MultipleParentClassLoader.Builder()
                               .append(features.mockedType)
                               .append(features.interfaces)
-                              .append(MultipleParentClassLoader.class.getClassLoader())
                               .append(Thread.currentThread().getContextClassLoader())
-                              .build(), ClassLoadingStrategy.Default.INJECTION)
+                              .append(MockAccess.class, DispatcherDefaultingToRealMethod.class)
+                              .append(MockMethodInterceptor.class,
+                                      MockMethodInterceptor.ForHashCode.class,
+                                      MockMethodInterceptor.ForEquals.class).build(), ClassLoadingStrategy.Default.INJECTION)
                       .getLoaded();
     }
 

